### PR TITLE
Fixed #25688 -- Made admin.register() disallow an empty list of models.

### PR DESCRIPTION
**Pull request by hramezani**
admin.register shouldn't allow an empty list of models to be registered

### CODE REVIEW

File: django/contrib/admin/decorators.py [hunk 1]
```diff
@@ -3,6 +3,8 @@ def register(*models, **kwargs):
     Registers the given model(s) classes and wrapped ModelAdmin class with
     admin site:
 
+    Atleast one model must be passed for register.
```
Comment (timgraham) [inline django/contrib/admin/decorators.py]: At least (two words)
"to register" (instead of "for")

File: django/contrib/admin/decorators.py [hunk 2]
```diff
@@ -14,6 +16,9 @@ class AuthorAdmin(admin.ModelAdmin):
     from django.contrib.admin.sites import site, AdminSite
 
     def _model_admin_wrapper(admin_class):
+        if not models:
+            raise ValueError('Atleast one model must be passed for register')
```
Comment (timgraham) [inline django/contrib/admin/decorators.py]: comments as above + add period.

**Comment by hramezani** [comment]
@timgraham, Thanks for help. I fixed comment and error messages.

**Comment by hramezani** [comment]
test added

**Comment by hramezani** [comment]
@timgraham is there any problem?

File: django/contrib/admin/decorators.py [hunk 3]
```diff
@@ -14,6 +16,9 @@ class AuthorAdmin(admin.ModelAdmin):
     from django.contrib.admin.sites import site, AdminSite
 
     def _model_admin_wrapper(admin_class):
+        if not models:
+            raise ValueError('At least one model must be passed to register')
```
Comment (timgraham) [inline django/contrib/admin/decorators.py]: Please add a period (I guess the test doesn't pass).

File: tests/admin_registration/tests.py [hunk 4]
```diff
@@ -133,3 +133,7 @@ def test_wrapped_class_not_a_model_admin(self):
     def test_custom_site_not_an_admin_site(self):
         self.assertRaisesMessage(ValueError, 'site must subclass AdminSite',
             register(Person, site=Traveler), NameAdmin)
+
+    def test_empty_models_list_registration_fails(self):
+        self.assertRaisesMessage(ValueError, 'At least one model must be passed to register.',
```
Comment (timgraham) [inline tests/admin_registration/tests.py]: use the context manager version: `with self.assertRaisesMessage(...):`

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by hramezani** [comment]
@timgraham 
period added to error message.
context manager version used.

**Comment by timgraham** [comment]
merged in 8c553e7d3fd3f598a445f3c35711829e9b3645a4, thanks!

**Pull request closed without merging**